src/Fmt_ast.ml

```diff
@@ -3710,16 +3710,47 @@ and fmt_structure c ctx itms =
           | Pmod_ident _ -> true
           | _ -> false
         in
-        let is_simple itm =
-          match itm.pstr_desc with
-          | Pstr_include {pincl_mod= me} | Pstr_module {pmb_expr= me} ->
-              is_simple_mod me
-          | Pstr_open _ -> true
-          | _ -> false
+        let is_simple (itm, c) =
+          match c.conf.structure_item_grouping with
+          | `Compact ->
+              Location.width itm.pstr_loc <= c.conf.margin
+              && Location.is_single_line itm.pstr_loc
```

Comment:
Using `is_simple_mod` here is probably too specific, it seems better to check that the location is narrow and on one line to catch all cases where there is a force-break in `itm`.

Reply:
It seems good to me, thank you.